numpy/linalg/linalg.py

```diff
@@ -2271,6 +2271,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if x.size == 0:
+        return 0
```

Comment:
This check is too early - it allows `np.linalg.norm([], axis="I have no idea what I'm doing")`, which should clearly error